tests/components/mqtt/test_discovery.py

```diff
@@ -1454,7 +1454,7 @@ class TestFlow(config_entries.ConfigFlow):
 
         async def async_step_mqtt(self, discovery_info: MqttServiceInfo) -> FlowResult:
             """Test mqtt step."""
-            return self.async_abort(reason="already_configured")
+            return self.async_abort(reason="single_instance_allowed")
```

Comment:
Do we have other tests which checks we don't abort if the abort reason is different?

Reply:
Not yet, we only had a test for `already_configured` which I adjusted.
I'll add a test for the `already_configured` to prove the subscription remains.

Reply:
Parameterized the tests now, and added some cases where we do not unsubscribe